src/interface/src/app/account/edit-field/edit-field.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { User } from '../../types';
+import { take } from 'rxjs';
+import { AuthService } from '../../services';
+
+type AvailableUserFields = keyof Pick<User, 'firstName' | 'lastName'>;
+type State = 'view' | 'editing' | 'saving' | 'error';
+
+@Component({
+  selector: 'app-edit-field',
+  templateUrl: './edit-field.component.html',
+  styleUrls: ['./edit-field.component.scss'],
+})
+export class EditFieldComponent {
```

Comment:
Small nit: At first read, I thought you had made a big component that just wraps the mat-form-field, not a form, so I wonder if we might consider renaming this to something like UserFieldEditorForm or something that makes it clear this isn't a form field.

I know that renaming a component can be a pain, so this is definitely not a blocker.

Reply:
thats a very good point! Will do